source/opt/loop_unroller.cpp

```diff
@@ -348,20 +386,44 @@ void LoopUnrollerUtilsImpl::PartiallyUnrollResidualFactor(ir::Loop* loop,
   AddBlocksToFunction(loop->GetMergeBlock());
   blocks_to_add_.clear();
 
+  // Create a new merge block for the first loop.
   InstructionBuilder builder{context_, new_exit_bb_raw};
   // Make the first loop branch to the second.
   builder.AddBranch(new_loop->GetHeaderBlock()->id());
 
   loop_condition_block_ = state_.new_condition_block;
   loop_induction_variable_ = state_.new_phi;
-
   // Unroll the new loop by the factor with the usual -1 to account for the
   // existing block iteration.
   Unroll(new_loop, factor);
 
+  LinkLastPhisToStart(new_loop);
+  AddBlocksToLoop(new_loop);
+
+  // Add the merge block to the back of the binary.
```

Comment:
back of the binary?  I don't understand the comment.

Reply:
It's a allusion to the fact that the blocks in the function are in eventual binary order so we have to be careful when we insert the new merge block to make sure it is at the end of the new loop list as to not break the dominator ordering rule. I'll clear up the comment